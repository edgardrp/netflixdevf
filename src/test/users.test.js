const { graphql } = require('graphql');
const { schema } = require('../index');
const setUpTest = require('./helpers');
const mutationRegister = `

    mutation Register($first_name:String!, $last_name:String!, $email:String!, $password:String!) {
        signup(data:{first_name:$first_name, 
            last_name:$last_name, 
            email:$email,
            password:$password
        }){
            token
        }
    }

`

describe('Register user works correctly', () => {

    beforeEach(async () => await setUpTest());

    it('Should create user correctly', async () => {
        const first_name = "Prueba";
        const last_name = "Prueba";
        const email = "prueba@prueba.com";
        const password = "prueba";

        const res = await graphql(
            schema,
            mutationRegister,
            null,
            {},
            { first_name, last_name, email, password }
        );

        expect(res).toMatchSnapshot();
        expect(res).toHaveProperty("data");
    });

});