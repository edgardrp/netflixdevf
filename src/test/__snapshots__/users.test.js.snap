// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register user works correctly Should create user correctly 1`] = `
Object {
  "data": Object {
    "signup": Object {
      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjViZjhkN2E1ZDllMTkyMzRlYTYxYTlkMyIsImVtYWlsIjoicHJ1ZWJhQHBydWViYS5jb20iLCJmaXJzdF9uYW1lIjoiUHJ1ZWJhIiwiaWF0IjoxNTQzMDM0NzkwfQ.5JOl6VumNUUOejyVNz1IXYiqOEtIpndrDa_omYpro38",
    },
  },
}
`;
